src/components/baklava-options-plugin/DragNumberOption.vue

```diff
@@ -16,46 +14,75 @@
         type="number"
         class="dark-input"
         :class="{ '--invalid': invalid }"
-        style="text-align: right;"
+        style="text-align: right; -moz-appearance: textfield;"
         @blur="leaveEditMode"
         @keydown.enter="leaveEditMode"
-        @mousedown="clicking = true"
       />
     </div>
-    <div class="__button --inc" @click="increment">
+    <div class="__button --inc" @click="incrementVal()">
       <i-arrow></i-arrow>
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import { Watch } from "vue-property-decorator";
+import { Watch, Component } from "vue-property-decorator";
 import { NumberOption } from "@baklavajs/plugin-options-vue";
-// import Arrow from "@baklavajs/plugin-options-vue/dist/baklavajs-plugin-options-vue/src/Arrow.vue.d"
-
-// @Component({
-//   components: {
-//     "i-arrow": Arrow,
-//   },
-// })
+import DragIntegerOption from "./DragIntegerOption.vue";
 
+/**
+ * Superclass for DragIntegerOption and DragFloatOption
+ */
+@Component
 export default class DragNumberOption extends NumberOption {
-  clicking = false;
+  displayValue = 0;
+
+  @Watch("editMode")
+  updateVals() {
+    this.displayValue = this.v;
+  }
 
-  // @Model('change', { type: Number }) readonly clicking!: boolean
+  handleMouseMove(e: MouseEvent) {
+    const step = this.option.step || 1;
+    this.displayValue += e.movementX * step;
+  }
+
+  created() {
+    if (this.option.defaultValue) this.displayValue = this.option.defaultValue;
```

Comment:
this.value (the prop) gets set with the right value at some point in time. Idk if it's at  `mounted()` or `created()`, but Baklava already has that built in, so let's use it.